### DISCUSSION TITLE
Create blade file for each PowerGrid table component

### DISCUSSION THREAD

**Discussion by aymanalhattami** [Power-Components/livewire-powergrid#607] [Ideas]
# ⚡ PowerGrid ⚡ Feature Request

## Summary
First thanks a lot for your efforts in developing this package and make it public. It is fabulous.

------------------------------------------------------
When creating new PowerGrid table, What I need is to be able to customize each blade file of PowerGrid table independently 
Ex: add new components, change some style, add some logic

It is better to create new blade file for each PowerGrid table with default content from base PowerGrid blade file

**Comment by akulmehta** [comment]
You can publish the views using

```
php artisan vendor:publish --tag=livewire-powergrid-views
```

Then you can edit the views to customize them.

**Reply by fahedtms** [reply]
@akulmehta
I did that and tried to customize the views but its not working, if you have an idea that would be great

**Reply by akulmehta** [reply]
If you can share some code I may be able to.

**Reply by fahedtms** [reply]
![image](https://user-images.githubusercontent.com/103887694/199009095-d2a548e1-4d29-4d77-a9e2-cf6965b9a379.png)

I just publish the views and trying to customize the select component but nothing changing. even tried to do 'dd()' its seems like its not going to the published vendor

**Reply by fahedtms** [reply]
what I am trying to achieve is to make the arrow to the left

![image](https://user-images.githubusercontent.com/103887694/199009959-28a99a34-c80f-4de7-9739-7c7f7da8867c.png)

**Reply by akulmehta** [reply]
Are you trying to change the position of the dropdown arrow for the number of items per page? If yes, then I don't think it is in `select.blade.php`.

**Reply by fahedtms** [reply]
for the select in the filters 

![image](https://user-images.githubusercontent.com/103887694/199078811-0cc1537e-a8f0-4057-9527-3eecaa9f1171.png)

**Reply by akulmehta** [reply]
I am confused - above your screenshot shows the dropdown for the number of items, but below you have it for select. Please check and decide what part of the table do you want edited. Different components are present in different blade files.

**Reply by fahedtms** [reply]
@akulmehta 
its obvious that I am implementing RTL version so I need to do it for all
what I am showing you is in the select that's what I am trying and didn't work

**Comment by aymanalhattami** [comment]
@akulmehta 

Publishing views allow to make one customization for all table views at one place, but I want to make different customization for each table view 

How can I do that?

**Comment by akulmehta** [comment]
Maybe you can define a variable in the table php file, and then look for the variable in the view and base conditionals off of that.

**Comment by aymanalhattami** [comment]
> Maybe you can define a variable in the table php file, and then look for the variable in the view and base conditionals off of that.

Good idea ☝

Is it possible to make this feature if I forked this package ?

**Comment by akulmehta** [comment]
You do not need to fork the package. You can define new variables in the php class. Publish the views as stated before. Then change the views as per your needs.
